### PR TITLE
Ti/alignment

### PR DESCRIPTION
**Pull request by SergejMuhic**
1. created the first draft for alignment TI based on requirements analysis of IFC Bridge

### CODE REVIEW

File: IFC4x3/ModelViews/TI-Bridge/DocModelView.xml [hunk 1]
```diff
@@ -1,6 +1,47 @@
...
 <DocModelView xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="TI-Bridge" UniqueId="14849e45-e262-4cfc-a27c-2bf98308d8f3" BaseView="e5d27119-bcb5-43e6-b6b3-8ababc4593f4">
 	<ConceptRoots>
+		<DocConceptRoot Name="Alignment" UniqueId="75c3c532-ca41-4dfd-befc-56d2265421db">
+			<ApplicableEntity xsi:type="DocEntity" xsi:nil="true" href="IfcAlignment" />
+			<Concepts>
+				<DocTemplateUsage Name="Name" UniqueId="ccf142e0-fe82-4307-b53b-dc6b210a0604">
+					<Documentation>Name of the alignment used for referencing purposes.</Documentation>
+					<Definition xsi:type="DocTemplateDefinition" xsi:nil="true" href="Object_User_Identity_31di66dFr7AJKdxDG2D1A" />
+					<Items>
+						<DocTemplateItem UniqueId="3aa5783a-10bd-465c-b2ef-abbe2e73317a" RuleParameters="ObjectName=All datasets name alignment &apos;Alignmemt&apos;;" />
```
Comment (pjanck) [inline IFC4x3/ModelViews/TI-Bridge/DocModelView.xml]:
```suggestion
						<DocTemplateItem UniqueId="3aa5783a-10bd-465c-b2ef-abbe2e73317a" RuleParameters="ObjectName=All datasets name alignment &apos;Alignment&apos;;" />
```

File: IFC4x3/ModelViews/TI-Bridge/DocModelView.xml [hunk 2]
```diff
@@ -1,6 +1,47 @@
...
+					</Items>
+				</DocTemplateUsage>
+				<DocTemplateUsage Name="AxisGradientCurve" UniqueId="237cbb24-fc03-4418-b7e5-95f2141cb971">
+					<Documentation>Alignment shall define an axis curve, which may consist of a horizontal alignment curve only, a horizontal curve and vertical curve.</Documentation>
```
Comment (pjanck) [inline IFC4x3/ModelViews/TI-Bridge/DocModelView.xml:15]:
```suggestion
					<Documentation>Alignment shall define an axis curve, which may consist of a horizontal alignment curve only, or both a horizontal and a vertical curve.</Documentation>
```

File: IFC4x3/ModelViews/TI-Bridge/DocModelView.xml [hunk 3]
```diff
@@ -1,6 +1,47 @@
...
+					<Items>
+						<DocTemplateItem UniqueId="3f5e9227-8bf9-40a2-aeea-9610a41f4cb4" />
+					</Items>
+				</DocTemplateUsage>
+				<DocTemplateUsage Name="AxisPolyline" UniqueId="05c11688-a64b-46cc-8053-3c7931c44565">
+					<Documentation>Alignment shall define an axis curve, which may consist of a polyline for capturing existing conditions at approximate intervals.</Documentation>
+					<Definition xsi:type="DocTemplateDefinition" xsi:nil="true" href="Axis_2D_Geometry_1u94Jere57Mf0qgUVrF6rb" />
+					<Items>
+						<DocTemplateItem UniqueId="f34127f9-c7d9-4f2e-b362-125557948bd0" RuleParameters="Items=IfcPolyline;" />
+					</Items>
```
Comment (pjanck) [inline IFC4x3/ModelViews/TI-Bridge/DocModelView.xml:26]: I assume that the `IfcPolyline` could be 3D. Propose to use Axis 3D Geometry instead.

File: IFC4x3/ModelViews/TI-Bridge/DocModelView.xml [hunk 4]
```diff
@@ -1,6 +1,47 @@
...
+					<Items>
+						<DocTemplateItem UniqueId="89806e87-48c0-4a6c-8da8-ca14e6b25899" RuleParameters="Items=IfcOffsetCurveByDistances;" />
+					</Items>
+				</DocTemplateUsage>
+				<DocTemplateUsage Name="Placement" UniqueId="dcc6f16b-d065-44fe-b078-26e76ed879a6">
+					<Documentation>Alignment shall be placed at the global origin.</Documentation>
+					<Definition xsi:type="DocTemplateDefinition" xsi:nil="true" href="Product_Local_Placement_3Bw5jVUH9AGukt7cEGB9j" />
+					<Items>
+						<DocTemplateItem UniqueId="5db44495-e210-4e9a-85f4-53b99927bda7" />
+					</Items>
```
Comment (pjanck) [inline IFC4x3/ModelViews/TI-Bridge/DocModelView.xml]: Propose to add rule: `IfcAlignment.ObjectPlacement.PlacementRelTo = NIL`. I am unsure about the syntax within IfcDoc.

**Review by pjanck — CHANGES_REQUESTED** [review]

File: IFC4x3/ModelViews/TI-Bridge/DocModelView.xml (hunk 1 above)
Comment (SergejMuhic) [reply IFC4x3/ModelViews/TI-Bridge/DocModelView.xml]: I will change this template a bit.

File: IFC4x3/ModelViews/TI-Bridge/DocModelView.xml (hunk 2 above)
Comment (SergejMuhic) [reply IFC4x3/ModelViews/TI-Bridge/DocModelView.xml:15]: Not changing this yet. Requirements are from bridge RA.

File: IFC4x3/ModelViews/TI-Bridge/DocModelView.xml (hunk 3 above)
Comment (SergejMuhic) [reply IFC4x3/ModelViews/TI-Bridge/DocModelView.xml:26]: I think I should add it to both. It is not clear in the requirements.

File: IFC4x3/ModelViews/TI-Bridge/DocModelView.xml (hunk 4 above)
Comment (SergejMuhic) [reply IFC4x3/ModelViews/TI-Bridge/DocModelView.xml]: Looks like I have to add that feature. Cannot find that option. Maybe it serialized correctly in mvdxml already. Have to check that first.

Comment (SergejMuhic) [reply IFC4x3/ModelViews/TI-Bridge/DocModelView.xml]: Nope, feature to add.

Comment (pjanck) [reply IFC4x3/ModelViews/TI-Bridge/DocModelView.xml]: As a `TemplateRule.Parameter` in mvdXML Grammar, using [Product Local Placement](https://bsi-infraroom.github.io/IFC-Documentation-Tunnel/4_4_0_0/general/HTML/link/product-local-placement.htm) concept template:

> RelativeToElement[EXISTS]=FALSE

Comment (pjanck) [reply IFC4x3/ModelViews/TI-Bridge/DocModelView.xml]: Given the proposal from #1, this would look like:

> RelativeToElement:NOTEXISTS

**Review by pjanck — APPROVED** [review]
There are merge conflicts that should be addressed before merging. Otherwise LFTM.

**Pull request merged**